app/discovery/discovery.go

```diff
@@ -146,6 +148,37 @@ func (s *Service) Match(srv, src string) (string, MatchType, bool) {
 	return src, MTProxy, false
 }
 
+// ScheduleHealthCheck starts background loop with health-check
+func (s *Service) ScheduleHealthCheck(ctx context.Context, interval time.Duration) {
+	log.Printf("health-check scheduled every %s", interval)
+
+	go func() {
+	hloop:
+		for {
+			timer := time.NewTimer(interval)
```

Comment:
the timer inside the loop bothers me. Maybe you meant a ticker created outside of the loop?